File: skeleton/src/me/shkschneider/skeleton/security/SimpleCrypt.java
package me.shkschneider.skeleton.security;

import org.jetbrains.annotations.NotNull;

public class SimpleCrypt {

    private static final String ALGORITHM = "XOR";

    private char[] mKey;

    public SimpleCrypt(@NotNull final String key) {
        mKey = key.toCharArray();
    }

    public String algorithm() {
        return ALGORITHM;
    }

    public String key() {
        return new String(mKey);
    }

    public String encrypt(@NotNull final String string) {
        final char[] chars = string.toCharArray();
        final char[] encrypted = new char[chars.length];
        for (int i = 0; i < chars.length; i++) {
            encrypted[i] = (char) (chars[i] ^ mKey[i % mKey.length]);
        }
        return Base64Helper.encrypt(new String(encrypted).getBytes());
    }

    public String decrypt(@NotNull final String string) {
        char[] encrypted = Base64Helper.decrypt(string).toCharArray();
        char[] decrypted = new char[encrypted.length];
        for (int i = 0; i < encrypted.length; i++) {
            decrypted[i] = (char) (encrypted[i] ^ mKey[i % mKey.length]);
        }
        return new String(decrypted);
    }

    // TODO

}

File: library/src/main/kotlin/me/shkschneider/skeleton/security/SimpleCrypt.kt
package me.shkschneider.skeleton.security

class SimpleCrypt : ICrypt<String> {

    private val ALGORITHM = "XOR"

    private val key: CharArray

    constructor(key: String) : super(key) {
        this.key = key.toCharArray()
    }

    override fun algorithm(): String {
        return ALGORITHM
    }

    override fun key(): String {
        return String(key)
    }

    override fun encrypt(src: String): String? {
        val chars = src.toCharArray()
        val encrypted = CharArray(chars.size)
        for (i in chars.indices) {
            encrypted[i] = (chars[i].toInt() xor key[i % key.size].toInt()).toChar()
        }
        return Base64Helper.encode(String(encrypted).toByteArray())
    }

    override fun decrypt(src: String): String? {
        val encrypted = Base64Helper.decode(src)
        val decrypted = CharArray(encrypted.size)
        for (i in encrypted.indices) {
            decrypted[i] = (encrypted[i].toInt() xor key[i % key.size].toInt()).toChar()
        }
        return String(decrypted)
    }

}
